_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -116,15 +105,17 @@ mfxStatus mfxSchedulerCore::Initialize2(const MFX_SCHEDULER_PARAM2 *pParam)
 
             auto thread_proc = [this](MFX_SCHEDULER_THREAD_CONTEXT* ctx) { ThreadProc(ctx); };
             // start threads
-            for (i = 0; i < m_param.numberOfThreads; ++i)
+            for (i = 0; i < m_param.params.NumThread; ++i)
             {
```

Comment:
It's stated that sessions are started without any threads but here we're creating at least 2 threads.

Reply:
no, we don't. params.NumThread is mediasdk public uAPI given to the user to specify on MFXInitEx how many threads he wants to have. If he used the API we will create this number of threads and ignore everything which components said to us. However, by default (when user did not use the uAPI) we will get param.NumThread=0 here and create nothing.